client/src/components/UI/Input/Input.tsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { CustomInputProps } from '../../../types/form.interface';
+
+const CustomInput: React.FC<CustomInputProps> = ({
```

Comment:
instead of building your own form inputs, have you guys talked about using a prebuilt component library? it might save you some time.

Reply:
We didn't discuss it. Basically, I was building the frontend part by myself.
I had custom form elements in previous projects, so I just copied from there.

Reply:
since you're taking the lead on the frontend, it's your decision. but if i was an interviewer, i'd ask why you did or did not chose to use a component library. What are the pros/cons of that choice?

Reply:
The main reason why I chose it is just to reduce the bundle size and don't overload the app with unnecessary libraries, since I already have premade reusable custom form components. As far as I know, these libraries can be significant in size.
Also, as we scale, we might run into some customization limitations as well as styling problems. 
On the other side, premade libraries can be a better option if we need to make an app quickly and don't care about design and customization.
I would go with the component library if I didn't have pre-made components.

Reply:
cool. are you familiar with tree shaking? When you use a bundler doesn't it filter out the code that you're not using to reduce the bundle size?

Reply:
I'd heard of it, but I wasn't particularly familiar with it.
I thought that React didn't use it until you connected bundlers like Webpack.

But I just did some research, and it turned out that Next.js uses Webpack under the hood as a bundler, and tree shking is already inculded when we build the application.
So I think the main arguments are that flexibility with customization and styling, as well as opportunities for scale, should not depend on third-party components.